### ISSUE TITLE
Addd a "debug" mode 

### ISSUE THREAD

**Issue by rollandf** [RedHat-Israel/ROSE#262]
A  "debug" mode of running the race will be helpful for students to help them track issues and improve.

The idea is to have an additional web page with different controls, where the user will not run the whole race but will be able to go step by step.
The proposed controls : 
- Start race in debug mode
- Stop race
- Run step

**Comment by nirs** [comment]
I don't think we need another web page, only different buttons above the dashboard, and maybe more info on the dashboard?

However, are you sure it will be useful to step manually 60 times instead of clicking start once? sounds like lot of work for the users.

**Comment by rollandf** [comment]
What about a "pause/resume" button, with the option of a "Step" action when in "Pause" state ?

**Comment by nirs** [comment]
Sounds good!

On the server side, this should be easy:
- Pause stop the looper
- Step call tick() (same method called by the looper)
- Resume start the looper

On the UI side:
1. Click "Start"
2. Click "Pause"
3. Click "Step" several times
4. Click "Resume"

I think we should build a quick prototype and see how it feels.
